### PR TITLE
Add a new plugin for C#

### PR DESCRIPTION
**Pull request by kurmeroli**
Hi there,

I've tried to make a  C# plugin for the graphql-code-generator. The generated code works fine with the official GraphQL.NET library.
Honestly the code might not be that pretty everywhere. But it's a first running version. And we thought it would be a pity doing this and don't share it as open source.

### CODE REVIEW

File: packages/plugins/c-sharp/apollo-dotnet/src/visitor.ts [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  }[] = [];
+
+  constructor(schema: GraphQLSchema, fragments: LoadedFragment[], rawConfig: ApolloDotNetRawPluginConfig, documents?: Types.DocumentFile[]) {
+    super(
+      schema,
+      fragments,
+      rawConfig,
+      {
+        sdkClass: rawConfig.sdkClass,
+        ngModule: rawConfig.ngModule,
```
Comment (ardatan) [inline packages/plugins/c-sharp/apollo-dotnet/src/visitor.ts]: Do we need that?

**Comment by ardatan** [comment]
Thank you @kurmeroli ! Amazing work.
Could you check my comment ?

File: packages/plugins/c-sharp/apollo-dotnet/src/visitor.ts (hunk 1 above)
Comment (kurmeroli) [reply packages/plugins/c-sharp/apollo-dotnet/src/visitor.ts:42]: maybe not, I will check this and let you know once I finished.
sorry not sure how early I could have a look on it. maybe during the easter weekend

Comment (kurmeroli) [reply packages/plugins/c-sharp/apollo-dotnet/src/visitor.ts]: we did not need that, it has been removed.

Comment (ardatan) [reply packages/plugins/c-sharp/apollo-dotnet/src/visitor.ts]: Thank you! Could you align graphql version as well? So we can get it merged.

Comment (kurmeroli) [reply packages/plugins/c-sharp/apollo-dotnet/src/visitor.ts]: aligned to graphql 15.0.0

**Comment by ardatan** [comment]
LGTM @dotansimha

**Comment by dotansimha** [comment]
Thanks @kurmeroli !

**Comment by clintsinger** [comment]
Hi, this is a great help. 
 
One thing I noticed is that it doesn't do Pascal Case even though that is supposed to be the default.   Shouldn't it upper case the first letter of each field? 

It doesn't even do it if explicitly set in the codegen configuration file.

**Pull request merged**